types/applications/filePicker.d.ts

```diff
@@ -0,0 +1,455 @@
+  protected _onDragStart(event: DragEvent): void;
+
+  /**
+   * @override
+   */
+  protected _onDrop(event: DragEvent): Promise<FilePicker.Result | void>;
+
+  /**
+   * Handle user submission of the address bar to request an explicit target
+   * @param event - The originating keydown event
+   */
+  protected _onRequestTarget(event: KeyboardEvent): void;
+
+  /**
+   * Handle requests from the IntersectionObserver to lazily load an image file
+   */
+  protected _onLazyLoadImages(...args: any): any; // TODO: type this when SidebarTab is typed
+
+  /**
+   * Handle file or folder selection within the file picker
+   * @param event - The originating click event
+   */
+  protected _onPick(event: Event): Promise<FilePicker.Result> | void;
+
+  /**
+   * Handle backwards navigation of the fol6der structure
+   */
+  protected _onClickDirectoryControl(
+    event: Event
+  ):
+    | ReturnType<this['browse']>
+    | Promise<ReturnType<this['browse']> | void | null>
+    | ReturnType<typeof FilePicker['configurePath']>;
+
+  /**
+   * Present the user with a dialog to create a subdirectory within their currently browsed file storate location.
+   */
+  protected _createDirectoryDialog(source: FilePicker.Source): Promise<ReturnType<this['browse']> | void | null>;
+
+  /**
+   * Handle changes to the bucket selector
+   */
+  protected _onChangeBucket(event: Event): ReturnType<this['browse']>;
+
+  /**
+   * @param event - (unused)
+   * @override
+   */
+  protected _onSearchFilter(event: any, query: string, html: HTMLElement): void;
+
+  /**
+   * Handle file picker form submission
+   */
+  protected _onSubmit(ev: Event): void;
+
+  /**
+   * Handle file upload
+   */
+  protected _onUpload(ev: Event): Promise<void>;
+
+  /**
+   * Bind the file picker to a new target field.
+   * Assumes the user will provide a <button> HTMLElement which has the data-target and data-type attributes
+   * The data-target attribute should provide the name of the input field which should receive the selected file
+   * The data-type attribute is a string in ["image", "audio"] which sets the file extensions which will be accepted
+   *
+   * @param button  - The button element
+   * @param options - (unused)
+   */
+  static fromButton(button: HTMLButtonElement, options?: any): FilePicker;
+
+  /**
+   * Record the last-browsed directory path so that re-opening a different FilePicker instance uses the same target
+   * @defaultValue `''`
+   */
+  LAST_BROWSED_DIRECTORY: string;
+
+  /**
+   * Record the last-configured tile size which can automatically be applied to new FilePicker instances
+   * @defaultValue `null`
+   */
+  LAST_TILE_SIZE: number;
+
+  /**
+   * Record the last-configured display mode so that re-opening a different FilePicker instance uses the same mode.
+   * @defaultValue `'list'`
+   */
+  LAST_DISPLAY_MODE: string;
+
+  /**
+   * Enumerate the allowed FilePicker display modes
+   */
+  DISPLAY_MODES: ['list', 'thumbs', 'tiles', 'images'];
```

Comment:
This prevents users from extending this and adding additional display modes. To enable this, we could make this `string[]` and write the default as `@defaultValue`. Not sure if it's worth it though.

Reply:
If they extend this, they also need to implement the according behavior. At that point you have to extend the class already anyway.